packages/eslint-plugin-react-hooks/src/code-path-analysis/code-path-analyzer.js

```diff
@@ -0,0 +1,802 @@
+ */
+function isHandledLogicalOperator(operator) {
+  return operator === '&&' || operator === '||' || operator === '??';
+}
+
+/**
+ * Checks whether the given assignment operator is a logical assignment operator.
+ * Logical assignments are taken into account for the code path analysis
+ * because of their short-circuiting semantics.
+ * @param {string} operator The operator found in the AssignmentExpression node
+ * @returns {boolean} `true` if the operator is "&&=" or "||=" or "??="
+ */
+function isLogicalAssignmentOperator(operator) {
+  return operator === '&&=' || operator === '||=' || operator === '??=';
+}
+
+/**
+ * Gets the label if the parent node of a given node is a LabeledStatement.
+ * @param {ASTNode} node A node to get.
+ * @returns {string|null} The label or `null`.
+ */
+function getLabel(node) {
+  if (node.parent.type === 'LabeledStatement') {
+    return node.parent.label.name;
+  }
+  return null;
+}
+
+/**
+ * Checks whether or not a given logical expression node goes different path
+ * between the `true` case and the `false` case.
+ * @param {ASTNode} node A node to check.
+ * @returns {boolean} `true` if the node is a test of a choice statement.
+ */
+function isForkingByTrueOrFalse(node) {
+  const parent = node.parent;
+
+  switch (parent.type) {
+    case 'ConditionalExpression':
+    case 'IfStatement':
+    case 'WhileStatement':
+    case 'DoWhileStatement':
+    case 'ForStatement':
+      return parent.test === node;
+
+    case 'LogicalExpression':
+      return isHandledLogicalOperator(parent.operator);
+
+    case 'AssignmentExpression':
+      return isLogicalAssignmentOperator(parent.operator);
+
+    default:
+      return false;
+  }
+}
+
+/**
+ * Gets the boolean value of a given literal node.
+ *
+ * This is used to detect infinity loops (e.g. `while (true) {}`).
+ * Statements preceded by an infinity loop are unreachable if the loop didn't
+ * have any `break` statement.
+ * @param {ASTNode} node A node to get.
+ * @returns {boolean|undefined} a boolean value if the node is a Literal node,
+ *   otherwise `undefined`.
+ */
+function getBooleanValueIfSimpleConstant(node) {
+  if (node.type === 'Literal') {
+    return Boolean(node.value);
+  }
+  return void 0;
+}
+
+/**
+ * Checks that a given identifier node is a reference or not.
+ *
+ * This is used to detect the first throwable node in a `try` block.
+ * @param {ASTNode} node An Identifier node to check.
+ * @returns {boolean} `true` if the node is a reference.
+ */
+function isIdentifierReference(node) {
+  const parent = node.parent;
+
+  switch (parent.type) {
+    case 'LabeledStatement':
+    case 'BreakStatement':
+    case 'ContinueStatement':
+    case 'ArrayPattern':
+    case 'RestElement':
+    case 'ImportSpecifier':
+    case 'ImportDefaultSpecifier':
+    case 'ImportNamespaceSpecifier':
+    case 'CatchClause':
+      return false;
+
+    case 'FunctionDeclaration':
+    case 'ComponentDeclaration':
```

Comment:
Since the changes to this package are trivial, what do you think about patching it instead? It wouldn't conflict that often and if it did it would be easy to fix.

Reply:
I mean as a git patch via https://www.npmjs.com/package/patch-package

Reply:
it's in `node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js` etc

Reply:
I'll leave that as a followup for someone who wants to figure this out. Not sure adding more hacks on top of the already complicated build infra is actually better than a few forked files.